### PR TITLE
Add un-deprecate project operation

### LINKED ISSUE #4507: Allow to undeprecate a project
**Motivation**
A project can be deprecated by mistake or people may change their mind.
Currently when this happens, it has to be patched in database which is also error prone

**Acceptance criteria**
1. An new endpoint allows to undeprecate a project
2. Resources can be created/updated again in this project

### PR DESCRIPTION
**Pull request by shinyhappydan**
Fixes #4507 

image

### CODE REVIEW

File: tests/src/test/scala/ch/epfl/bluebrain/nexus/tests/kg/ProjectsSpec.scala [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package ch.epfl.bluebrain.nexus.tests.kg
+
+import ch.epfl.bluebrain.nexus.tests.BaseIntegrationSpec
+
+class ProjectsSpec extends BaseIntegrationSpec {}
```
Comment (imsdu) [inline tests/src/test/scala/ch/epfl/bluebrain/nexus/tests/kg/ProjectsSpec.scala:5]: To be deleted

File: docs/src/main/paradox/docs/delta/api/projects-api.md [hunk 2]
```diff
@@ -185,6 +185,29 @@ Request
 
 Response
 :   @@snip [deprecated.json](assets/projects/deprecated.json)
+~~~~
+## Undeprecate
```
Comment (imsdu) [inline docs/src/main/paradox/docs/delta/api/projects-api.md:189]: We also need a paragraph in the release notes

File: delta/app/src/test/scala/ch/epfl/bluebrain/nexus/delta/routes/ProjectsRoutesSpec.scala [hunk 3]
```diff
@@ -534,11 +538,73 @@ class ProjectsRoutesSpec extends BaseRouteSpec with IOFromMap {
...
+      val (org, project) = thereIsAProject
+      Put(s"/v1/projects/$org/$project/undeprecate?rev=1") ~> as(superUser) ~> routes ~> check {
+        status shouldEqual StatusCodes.BadRequest
+        response.asJson shouldEqual jsonContentOf(
+          "projects/errors/project-not-deprecated.json",
+          "org"  -> org,
+          "proj" -> project
+        )
+      }
+    }
```
Comment (olivergrabinski) [inline delta/app/src/test/scala/ch/epfl/bluebrain/nexus/delta/routes/ProjectsRoutesSpec.scala:566]: Is it fair to say this does not check that the project is actually still active? It should also be possible to test this in `ProjectsImplSpec` instead and leave this as an error serialization test only

Comment (shinyhappydan) [reply delta/app/src/test/scala/ch/epfl/bluebrain/nexus/delta/routes/ProjectsRoutesSpec.scala:566]: Good spot :)

I was trying to avoid changing `ProjectsImplSpec` though as I was anticipating the same issues I had with this one. Do you think it's worth the investment?

**Review by imsdu — DISMISSED** [review]

File: delta/app/src/test/scala/ch/epfl/bluebrain/nexus/delta/routes/ProjectsRoutesSpec.scala (hunk 3 above)
Comment (olivergrabinski) [reply delta/app/src/test/scala/ch/epfl/bluebrain/nexus/delta/routes/ProjectsRoutesSpec.scala:569]: I don't think it needs to be done here. But in my experience the ImplSpec were easier to work with than the Route ones

Comment (imsdu) [reply delta/app/src/test/scala/ch/epfl/bluebrain/nexus/delta/routes/ProjectsRoutesSpec.scala:569]: Maybe something in the integration tests ?

Comment (shinyhappydan) [reply delta/app/src/test/scala/ch/epfl/bluebrain/nexus/delta/routes/ProjectsRoutesSpec.scala:569]: I purposely didn't add edge cases to the integration tests as they are so much slower, and now really better than testing at this level. The integration test has the "happy" path only, which seems like the correct way to test given that these route specs are almost end-to-end tests

**Review by imsdu — APPROVED** [review]
